### ISSUE TITLE
SMTP: [Errno 99] Cannot assign requested address

### ISSUE THREAD

**Issue by CodyManshack** [getsentry/sentry#20963]
## Important Details

How are you running Sentry?
**On-Premise w/ Docker, version 20.9.0**

## Description

I have configured Sentry to use the default SMTP server it deploys when running the `install.sh` command. My config is as follows:

> mail.backend: 'smtp'  # Use dummy if you want to disable email entirely
> mail.host: 'localhost'
> mail.port: 25
> mail.username: ''
> mail.password: ''
> mail.use-tls: false
> mail.from: 'noreply@sentry.mydomain.com'
> mail.enable-replies: false
> mail.reply-hostname: ''
> mail.mailgun-api-key: ''

No emails send from the system. The SMTP server never logs anything other than its initial startup and listen. However, the worker server does log each time an attempt to send an email is made:

```
error: [Errno 99] Cannot assign requested address
16:56:17 [ERROR] celery.app.trace: Task sentry.tasks.email.send_email[7c7b3f32-3692-4b46-b2ff-0bd3f7b0c9bf] raised unexpected: error(99, 'Cannot assign requested address') (data={u'internal': False, u'traceback': u'Traceback (most recent call last):\n  File "/usr/local/lib/python2.7/site-packages/celery/app/trace.py", line 412, in trace_task\n    R = retval = fun(*args, **kwargs)\n  File "/usr/local/lib/python2.7/site-packages/celery/app/trace.py", line 704, in __protected_call__\n    return self.run(*args, **kwargs)\n  File "/usr/local/lib/python2.7/site-packages/sentry_sdk/integrations/celery.py", line 171, in _inner\n    reraise(*exc_info)\n  File "/usr/local/lib/python2.7/site-packages/sentry_sdk/integrations/celery.py", line 166, in _inner\n    return f(*args, **kwargs)\n  File "/usr/local/lib/python2.7/site-packages/sentry/tasks/base.py", line 48, in _wrapped\n    result = func(*args, **kwargs)\n  File "/usr/local/lib/python2.7/site-packages/sentry/tasks/email.py", line 76, in send_email\n    send_messages([message])\n  File "/usr/local/lib/python2.7/site-packages/sentry/utils/email.py", line 408, in send_messages\n    sent = connection.send_messages(messages)\n  File "/usr/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 104, in send_messages\n    new_conn_created = self.open()\n  File "/usr/local/lib/python2.7/site-packages/django/core/mail/backends/smtp.py", line 64, in open\n    self.connection = self.connection_class(self.host, self.port, **connection_params)\n  File "/usr/local/lib/python2.7/smtplib.py", line 256, in __init__\n    (code, msg) = self.connect(host, port)\n  File "/usr/local/lib/python2.7/smtplib.py", line 317, in connect\n    self.sock = self._get_socket(host, port, self.timeout)\n  File "/usr/local/lib/python2.7/smtplib.py", line 292, in _get_socket\n    return socket.create_connection((host, port), timeout)\n  File "/usr/local/lib/python2.7/socket.py", line 575, in create_connection\n    raise err\nerror: [Errno 99] Cannot assign requested address\n', u'name': 'sentry.tasks.email.send_email', u'args': '()', u'kwargs': "{'message': <django.core.mail.message.EmailMultiAlternatives object at 0x7f744f6ffbd0>}", u'description': u'raised unexpected', u'hostname': u'celery@dbd50783283e', u'id': '7c7b3f32-3692-4b46-b2ff-0bd3f7b0c9bf', u'exc': "error(99, 'Cannot assign requested address')"})
16:58:11 [WARNING] sentry.tasks.release_registry: Release registry URL is not specified, skipping the task.
```

I attempted to change the `mail.host` key to '0.0.0.0' so that it would listen on all local devices, however that gave me `[Errno 111] Connection refused` errors, so I reverted that change back to 'localhost'.

## Steps to Reproduce

1. Install Sentry On Premise 20.9.0 with the default smtp configuration
2. Test your SMTP settings at the `manage/status/mail/` route
3. Observe error in bottom right of UI
4. `docker logs sentry_onpremise_smtp_1` - no errors
5. `docker logs sentry_onpremise_worker_1` - `[Errno 99] Cannot assign requested address` errors appear

### What you expected to happen

I should be able to send mail with the default configuration included with the ./install.sh script

**Comment by CodyManshack** [comment]
After further review, I've found that the `_web_` server is posting to the address: `127.20.0.1`.
![image](https://user-images.githubusercontent.com/36999949/94288538-cd4f8600-ff57-11ea-8b81-c275e26fa081.png)

I inspected the sentry network and found that not a single server was assigned that IP address.
![image](https://user-images.githubusercontent.com/36999949/94288565-d5a7c100-ff57-11ea-81aa-2fc395b52793.png)

In fact, it seems the lowest I could find was a .3. Lastly, I confirmed that nothing ever gets to `sentry_onpremise_smtp_1` - even if using a standard SMTP server like google or self-hosted in the config.

I'm not sure how relevant this is? Perhaps .1 is the docker gateway which does the NAT translation for the domain name? I don't know, but thought I would point this out as it seems the `_smtp_` never receives requests.
